projects/github-actions/test-results-to-slack/src/slack.js

```diff
@@ -54,6 +75,57 @@ function getBlocksChunks( blocks, chunkSize ) {
 	return chunks;
 }
 
+/**
+ * Split an array of blocks into chunks based on a given type property as delimiter
+ * E.g. if the array is [ {type: 'context'}, {type: 'context'}, {type: 'file'}, {type: 'context'} ] and the delimiter is 'file'
+ * the result will be [ [ {type: 'context'}, {type: 'context'} ], [ {type: 'file'} ], [ {type: 'context'} ] ]
+ *
+ * @param {[object]} blocks - the array to be split
+ * @param {string} type - the type property to use as delimiter
+ * @returns {[object]} the array of chunks
+ */
+function getBlocksChunksByType( blocks, type ) {
```

Comment:
I'm not sure I understand the purpose of this function. Do we need to group chunks before and after attachments were added to the blocks list? 

I wonder if, instead of splitting an array with a delimiter, we can `filter` attachments and other blocks into separate arrays and then sent attachment as needed.

Reply:
That would be easier, but it will break the order of the blocks. We want to send each screenshot right after the test name and error message, otherwise we won't understand much from them.

Reply:
Ah, yeah. Ok. Now I see that we have a single array of blocks for all the failures. It makes sense.

Just a thought for the future: We could introduce an object that encapsulates all the details for specific failure, including all the relevant metadata and attachments.